File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/api/ApiError.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.api;

import java.io.Serializable;
import java.util.List;

/**
 * The error from the API response body.
 */
public class ApiError implements Serializable {

    private static final long serialVersionUID = 8917688214136217124L;

    /**
     * The error ID
     */
    public String errorId;

    /**
     * The collection of errors, if this was returned from the server/API.
     */
    public List<ApiErrorItem> errors;
}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/api/ApiError.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.api

import java.io.Serializable

/**
 * The error from the API response body.
 */
class ApiError : Serializable {
    /**
     * The error ID
     */
    var errorId: String? = null

    /**
     * The collection of errors, if this was returned from the server/API.
     */
    var errors: List<ApiErrorItem>? = null

    companion object {
        @Suppress("Unused")
        private const val serialVersionUID = 8917688214136217124L
    }
}
